db/config.cc

```diff
@@ -99,6 +99,21 @@ error_injection_list_to_json(const std::vector<db::config::error_injection_at_st
     return value_to_json("error_injection_list");
 }
 
+template <>
+bool
+config_from_string(std::string_view value) {
+    // boost::lexical_cast doesn't accept true/false, which are our output representations
+    // for bools. We want round-tripping, so we need to accept true/false. For backward
+    // compatibility, we also accept 1/0. #19791.
+    if (value == "true" || value == "1") {
```

Comment:
Hmm `yaml` also [accepts](https://yaml.org/type/bool.html):
>  y|Y|yes|Yes|YES|n|N|no|No|NO
|true|True|TRUE|false|False|FALSE
|on|On|ON|off|Off|OFF

OTOH [json accepts](https://json-schema.org/understanding-json-schema/reference/boolean) only `true/false`, but why do we care about json? Our config file `scylla.yaml` comes only with the `*.yaml` extension.
Maybe using a specific parser instead of manually implementing its simplification will save us headaches in the future.

Reply:
config.cc mentions json. Maybe it's related to the REST API.

Our yaml parse is doesn't know about the types of the nodes it parses. I don't plan to replace the parse as part of getting booleans to round-trip in system.config.